pxtsim/localization.ts

```diff
@@ -0,0 +1,86 @@
+// Localization functions. Please port any modifications over to pxtlib/util.ts
+namespace pxsim.localization {
+    let _localizeStrings: Map<string> = {};
+
+    export function _localize(s: string) {
+        return _localizeStrings[s] || s;
+    }
+
+    export function setLocalizedStrings(strs: Map<string>) {
+        _localizeStrings = strs;
+    }
+
+    export function htmlEscape(_input: string) {
+        if (!_input) return _input; // null, undefined, empty string test
+        return _input.replace(/([^\w .!?\-$])/g, c => "&#" + c.charCodeAt(0) + ";");
+    }
+
+    export function jsStringQuote(s: string) {
+        return s.replace(/[^\w .!?\-$]/g,
+            (c) => {
+                let h = c.charCodeAt(0).toString(16);
+                return "\\u" + "0000".substr(0, 4 - h.length) + h;
+            });
+    }
+
+    export function fmt_va(f: string, args: any[]): string {
```

Comment:
I don't think we need to support all that stuff for the sim. Maybe we just simplify it to just swapping the strings.